node/index.js

```diff
@@ -11,16 +11,35 @@ const speechGoogle = require('./speech_google');
 const speechOpenAI = require('./speech_openai');
 const { state } = require('./state');
 
+// see versions here: https://github.com/wppconnect-team/wa-version/tree/main/html
+const wwebVersion = '2.2412.54';
+
+const puppeteerOptions = {
+        headless: true,
+        args: ['--no-sandbox', '--disable-setuid-sandbox']
+};
+
+if (env.chromiumPath) {
+    puppeteerOptions['executablePath'] = env.chromiumPath;
+}
+
+if (env.userAgent) {
+    puppeteerOptions.args.push(`--user-agent="${env.userAgent}"`);
+}
+
 // Setup options for the client and data path for the google chrome session
 const client = new Client({
     authStrategy: new LocalAuth({dataPath: env.chromeDataPath}),
-    puppeteer: {
-        headless: true,
-        args: ['--no-sandbox', '--disable-setuid-sandbox']
-    }
+    puppeteer: puppeteerOptions,
+    // Use a fixed, old webVersionCache to work with the latest WA version
+    // remove once whatsapp-web.js catches up.
+    // https://github.com/pedroslopez/whatsapp-web.js/issues/2789
+    webVersionCache: {
+        type: 'remote',
+        remotePath: `https://raw.githubusercontent.com/wppconnect-team/wa-version/main/html/${wwebVersion}.html`,
```

Comment:
if we basically never change the web version, can we instead include this HTML file into this repository?

Or do we plan to update the version number on a regular basis?

Reply:
I don't know exactly how this webcaching thing works and what is the time horizon for WA's support of older versions, so ideally we'd go back to use the default setting that is in WhatsappWeb.js when it supports the new versions.